### PR TITLE
Cached translators

### PR DESCRIPTION
**Pull request by Ydrasil**

### CODE REVIEW

**Comment by Ydrasil** [comment]
@vinyll Its hand tested on another project, if the file is already on the if condition, the second time it does not enter on the if condition.

File: tcii18n/template.py [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+
+_cached_translators = {}
+
+
+def flask_methods(app, get_translations_file):
+    @app.context_processor
+    def utility_processor():
+        def trans(sentence):
+            filename = get_translations_file()
+            if filename not in _cached_translators:
```
Comment (vinyll) [inline tcii18n/template.py]: can you check that this actually works?
You could verify that it is skipped if you call twice the same filename.
if it does not work, you should check with the key:

```
if filename not in _cached_translators.keys():
    …
```

Comment (vinyll) [inline tcii18n/template.py:11]: can you check that this actually works?
You could verify that it is skipped if you call twice the same filename.
if it does not work, you should check with the key:

```
if filename not in _cached_translators.keys():
    …
```

Comment (Ydrasil) [reply tcii18n/template.py:11]: I've already test it, and I confirm that this solution is working, I can also change with `if filename not in _cached_translators.keys():` if you prefer with way.

Comment (vinyll) [reply tcii18n/template.py:11]: If the it does check based on the key implicitly I'm fine with it

**Review by vinyll — APPROVED** [review]

**Pull request merged**